Chessington.GameEngine/Pieces/Pawn.cs

```diff
@@ -38,21 +38,31 @@ public override IEnumerable<Square> GetAvailableMoves(Board board)
             }
 
             // Pawns can take pieces diagonally
-            var diagonalLeft = new Square(location.Row + direction*1, location.Col - 1);
-            if (diagonalLeft.Col >= 0 && board.SquareIsOccupied(diagonalLeft) &&
-                board.GetPiece(diagonalLeft).Player != Player)
+            var diagonalLeft = GetDiagonalMoves(board, -1);
+            if (diagonalLeft != null) availableLocations.Add(diagonalLeft.Value);
+
+
+            var diagonalRight = GetDiagonalMoves(board, 1);
+            if (diagonalRight != null) availableLocations.Add(diagonalRight.Value);
+
+            return availableLocations;
+        }
+
+        private Square? GetDiagonalMoves(Board board, int horizontalDirection)
+        {
+            var location = board.FindPiece(this);
+            var verticalDirection = Player == Player.Black ? 1 : -1;
```

Comment:
You're calculating both of these twice (here and in the `GetAvailableMoves` method). Might be worth just passing them in to this method? 🙂

Reply:
`location` is `board` specific so I've done that, and I've popped `verticalDirection` out to be a property because it actually is a property of the `Pawn` object, all in 046c074. I hope that's ok 🙂